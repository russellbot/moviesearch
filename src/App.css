@font-face {
font-family: 'Vinyl repair kit';
font-style: normal;
font-weight: normal;
src: local('Vinyl repair kit'), url('VINYLREPAIRKIT.woff') format('woff');
}

h1 {
    font-family: 'Vinyl repair kit';
    font-weight: 200;
    color: #0ccac4;
}