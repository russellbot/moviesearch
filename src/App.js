import React, { Component } from 'react';
import CardList from './CardList';
import SearchBox from './SearchBox';
import './App.css';

class App extends Component {
    constructor() {
        super()
        this.state = {
            movies: [],
            searchfield: ''
        }
    }

    componentDidMount() {
        fetch('http://www.omdbapi.com/?s=star+wars&apikey=469942e0')
            .then(response=> response.json())
            .then(movie => this.setState({ movies: movie.Search}));
    }

    onSearchChange = (event) => {
        this.setState({ searchfield: event.target.value })
    }

    render() {
        const filteredMovies = this.state.movies.filter(movies => {
            return movies.Title.toLowerCase().includes(this.state.searchfield.toLowerCase());
        })
        return (
            <div className = 'tc'>
                <h1 className='f1'>star wars movie search</h1>
                <SearchBox searchChange={this.onSearchChange} />
                <CardList movies={filteredMovies} />
            </div>
        );
    }
}

export default App;