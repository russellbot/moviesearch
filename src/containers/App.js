import React, { Component } from 'react';
import { connect} from 'react-redux';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundry from '../components/ErrorBoundry';
import './App.css';

import { setSearchField } from '../actions.js';

const mapStateToProps = state => {
    return {
        searchField: state.searchField
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onSearchChange: (event) => dispatch(setSearchField(event.target.value))
    }
}

class App extends Component {
    constructor() {
        super()
        this.state = {
            movies: []
        }
    }

    componentDidMount() {
        fetch('https://www.omdbapi.com/?s=star+wars&apikey=469942e0')
            .then(response=> response.json())
            .then(movie => this.setState({ movies: movie.Search}));
    }

    render() {
        const { movies } = this.state;
        const { searchField, onSearchChange } = this.props;
        const filteredMovies = movies.filter(movie => {
            return movie.Title.toLowerCase().includes(searchField.toLowerCase());
        })
        if (movies.length === 0) {
            return <h1>Loading</h1>
        } else {
            return (
                <div className = 'tc'>
                    <h1 className='f1'>star wars movie search</h1>
                    <SearchBox searchChange={onSearchChange} />
                    <Scroll>
                        <ErrorBoundry>
                            <CardList movies={filteredMovies} />
                        </ErrorBoundry>
                    </Scroll>
                </div>
            );
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);