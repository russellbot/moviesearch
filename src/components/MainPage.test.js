import { shallow } from 'enzyme';
import React from 'react';
import App from './App';

it('expect to render App component', () => {
    const mockStore = {
        movies: [],
        searchField: ''
    }
    expect(shallow(<App store={mockStore} />)).toMatchSnapshot();
})