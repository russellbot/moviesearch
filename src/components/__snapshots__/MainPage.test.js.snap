// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        movies={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
